libs/common/src/change-stream/delete-document-with-audit-logs.test.ts

```diff
@@ -0,0 +1,109 @@
+import { when } from 'jest-when';
+import { ObjectId } from 'mongodb';
+import { MongoDbClient } from '../mongo-db-client';
+import { deleteDocumentWithAuditLogs } from './delete-document-with-audit-logs';
+import { generateNoUserLoggedInAuditDetails } from './generate-audit-details';
+import { MONGO_DB_COLLECTIONS } from '../constants';
+import { generateMockNoUserLoggedInAuditDatabaseRecord } from './test-helpers';
+
+const mockSession = {
+  withTransaction: jest.fn((callback: () => void) => {
+    callback();
+  }),
+  endSession: jest.fn(),
+};
+
+const mockClient = {
+  startSession: jest.fn(() => mockSession),
+};
+
+const mockGetCollection = jest.fn();
+
+const mockDeleteOne = jest.fn();
+const mockUsersCollection = {
+  deleteOne: mockDeleteOne,
+};
+
+const mockInsertOne = jest.fn();
+const mockDeletionsCollection = {
+  insertOne: mockInsertOne,
+};
+
+const mockDb = {
+  getClient: jest.fn(() => mockClient),
+  getCollection: mockGetCollection,
+} as unknown as MongoDbClient;
+
+describe('deleteDocumentWithAuditLogs', () => {
+  const documentId = new ObjectId();
+  const originalDeletionAuditLogsDeleteAfterSeconds = process.env.DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS;
+
+  beforeEach(() => {
+    process.env.DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS = '60';
+
+    jest.clearAllMocks();
+
+    when(mockGetCollection).calledWith('users').mockReturnValueOnce(mockUsersCollection);
+    when(mockGetCollection).calledWith(MONGO_DB_COLLECTIONS.DELETION_AUDIT_LOGS).mockReturnValueOnce(mockDeletionsCollection);
+  });
+
+  afterAll(() => {
+    process.env.DELETION_AUDIT_LOGS_DELETE_AFTER_SECONDS = originalDeletionAuditLogsDeleteAfterSeconds;
+  });
+
+  describe('when the insertion and deletion are successful', () => {
```

Comment:
could we have a when it fails -- and just check that we throw the expected error :) no need to worry about rollback -- thats part of mongo library

Reply:
nvm i think we cover this in our api tests -- gj